contents/staticAnalysis/ESLint.md

```diff
@@ -0,0 +1,65 @@
+
+ESLint is shipped with a wide range of [built-in rules](https://eslint.org/docs/rules/) that can be configured on installation. It also offers flexibility by allowing importing your own custom rules via [plugins](https://eslint.org/docs/developer-guide/working-with-plugins). ESLint is widely used by many organizations and has [editor integrations](https://eslint.org/docs/user-guide/integrations#editors) such as error highlighting in many popular editors like Atom and WebStorm.
+
+## How it is used
+
+After [installing with npm](https://eslint.org/docs/user-guide/getting-started#installation-and-usage), ESLint can be configured using [comments or a single configuration file](https://eslint.org/docs/user-guide/configuring#configuring-rules). You can learn how to set them up [here](https://eslint.org/docs/user-guide/configuring). ESLint allows you to disable rules for [specific files and directories](https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories) or for [specific lines](https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments).
+
+Once configured, you can lint your files with [terminal commands](https://eslint.org/docs/user-guide/command-line-interface) and you can add [options](https://eslint.org/docs/user-guide/command-line-interface#options) to the command based on your needs.
+
+ESLint can be easily integrated with your build tool of choice. Here are some [plugins](https://eslint.org/docs/user-guide/integrations#build-tools) for some popular build tools. ESLint can also work with Continuous Integration tools like Travis, AppVeyor and CircleCI. Simply add the ESLint command as your pretest/test script in `package.json`. Configuration is explained in this [example using Travis](https://medium.com/jason-feng/travis-ci-guide-bdc03b3dbc9d). You can also learn how to automate ESLint to [run every time you commit](https://medium.com/the-node-js-collection/why-and-how-to-use-eslint-in-your-project-742d0bc61ed7).
+
+## Rules
+ESLint comes shipped with a large number of [built-in rules](https://eslint.org/docs/rules/). They are simply added into your configuration file/comment and you can [configure the rules](https://eslint.org/docs/user-guide/configuring#configuring-rules) to suit your needs.
+
+An example of a rule would be [`curly`](https://eslint.org/docs/rules/curly). JavaScript allows for omission of curly braces when a block contains one statement. This rule enforces curly braces for blocks even when they are optional as it may lead to bugs and reduces code clarity.
+
+Configuration:
+```js
+// .eslintrc.js
+  rules: {
+    "curly": "error"
+  }
+```
+Additional options include "multi", "multi-line", "multi-or-nest". They can be configured in this way:
+
+```js
+// .eslintrc.js
+  rules: {
+    "curly": ["error", "multi"]
+  }
+```
+For the configuration above, it allows for blocks which span one line to omit the curly braces. eg:
+
+```js
+if (multi) thisPasses();
+```

```

Comment:
Here you are explaining _how_ to configure the rule, which is not in the scope of this chapter. Instead, explain _what_ can be configured, thereby showcasing the versatility of ESLint.
For example, you can give a sample js code that has three different variations of omitting curly braces and say that the cury rule can be configured to reject the first one, the first two, or all three of them.